fastcrypto-tbls/src/polynomial.rs

```diff
@@ -202,42 +223,38 @@ impl<C: GroupElement> Poly<C> {
             return Err(FastCryptoError::InvalidInput);
         }
 
-        let full_numerator =
-            C::ScalarType::product(indices.iter().map(|i| C::ScalarType::from(*i)));
-
-        let mut coeffs = Vec::new();
-        for i in &indices {
-            let mut negative = false;
-            let (mut denominator, remaining) = indices.iter().filter(|j| *j != i).fold(
-                (C::ScalarType::from(*i), 1u128),
-                |(prev_acc, remaining), j| {
-                    let diff = if i > j {
-                        negative = !negative;
-                        i - j
-                    } else {
-                        // i < j (but not equal)
-                        j - i
-                    };
-                    debug_assert_ne!(diff, 0);
-                    let either = Self::fast_mult(remaining, diff);
-                    match either {
-                        Either::Left((remaining_as_scalar, diff)) => {
-                            (prev_acc * remaining_as_scalar, diff)
-                        }
-                        Either::Right(new_remaining) => (prev_acc, new_remaining),
+        let full_numerator = C::ScalarType::product(
+            indices
+                .iter()
+                .map(|i| C::ScalarType::from(*i) - C::ScalarType::from(x)),
```

Comment:
x is fixed here, move this outside the loop

Reply:
Good point - that shouldn't happen. In both places it's called, that can never be the case, but I've added a check here to make it explicit.